### ISSUE TITLE
'Symbol' is undefined problem in ie11

### ISSUE THREAD

**Issue by lykhatskyi** [babel/babel#7597]
https://github.com/babel/babel/blob/e2c5f25e97757cf4f7bd1527c51a9b61a29115ac/packages/babel-runtime/helpers/builtin/iterableToArray.js#L2

getting an error in ie11, 
Can we change this check?
A lot of  clients are using different js validators that decrease  app rating because of this error
![image](https://user-images.githubusercontent.com/18104686/37607102-d2201926-2b9f-11e8-9aba-231340eb9a92.png)

**Comment by loganfsmyth** [comment]
Note the `builtin/` in the path of that file. That means you're passing [the `useBuiltins`](https://github.com/babel/babel/tree/v7.0.0-beta.0/packages/babel-plugin-transform-runtime#usebuiltins) option. Specifically

> When enabled, the transform will use helpers that do not use any polyfills from core-js.

Given that, it would be up to you to ensure that any necessary ES6 polyfills are available, or else avoid using syntax in Babel that would depend on ES6 functionality that hasn't been polyfilled.

**Comment by mrchief** [comment]
@loganfsmyth That sounds promising but doesn't solve the problem. 

![image](https://user-images.githubusercontent.com/781818/38900118-4f292b54-4268-11e8-956a-79245674050b.png)

**Comment by csergiu** [comment]
> avoid using syntax in Babel that would depend on ES6 functionality that hasn't been polyfilled.

then what's the purpose of Babel?

**Comment by mrchief** [comment]
> then what's the purpose of Babel?

Babel will do the polyfilling for you. But if you say `useBuiltins: true`, and try to use modern ES6 features in an archaic browser like Internet Explorer, then you'll run into issues. You'll be fine in Chrome, Safari, FF for most part but again, you need to ensure you're not using any new features that are not adopted by browsers yet.

As far as my comment above goes, I solved it by using babel-polyfill. I was using polyfill.io which was causing the `Symbol` related issues. As soon as I switched to babel-polyfill, all of them went away.

**Comment by ljharb** [comment]
I don't see why polyfilling `Symbol` should be required - it's not actually shimmable (core-js's shim is actually a sham, because `typeof` can't be polyfilled).

Most of the babel helpers have guards ensuring `Symbol` exists; why does `iterableToArray` lack one?

**Comment by ljharb** [comment]
That’s great; one will be incoming very soon!

**Comment by kofifus** [comment]
Stuck with the same issue :(
What is the best way to solve this until above commits become part of the release ?

**Comment by timtox** [comment]
@kofifus All you can do is rely on a workaround. I have the same problem where babel compiles `for (const cassette of cassettes)` to `Symbol.iterator`. As long we are waiting for the PR you can write `for (let i = 0; i < cassettes.length; i++)` which works on IE11.

**Comment by ljharb** [comment]
Our temporary workaround has been to manually patch the file generated pending the PR landing.

**Comment by nicolo-ribaudo** [comment]
@timtox You can use normal for-of and the `assumeArray` option of the `@babel/plugin-transform-for-of` plugin.

**Comment by timtox** [comment]
@nicolo-ribaudo Sorry for late response but thank you for the tip with the babel [plugin](https://www.npmjs.com/package/@babel/plugin-transform-for-of). Is there a list of ES6 features which are not baked into the babel core and one must use a plugin to transform them properly for older Browsers?

**Comment by ljharb** [comment]
@timtox every single one that adds syntax :-)

**Comment by timtox** [comment]
Thank you :-)

**Comment by johnnywang** [comment]
Was this never actually fixed? Started seeing this issue myself in a library that we can't actually run babel-polyfill for so the guard would be nice to have

**Comment by MatthewHerbst** [comment]
@johnnywang it's been stuck in PR: https://github.com/babel/babel/pull/8947

**Comment by TomS-** [comment]
Still an issue for me too. I've had this come up recently.

**Comment by Abo-Gabal** [comment]
i tried importing
core-js --> error still exist
react-app-polyfill --> error still exist
babel/polyfill --> error still exist

**Comment by johnnywang** [comment]
So looks like the PR died out and then was closed out because it was never approved. Is anyone else taking a look into this?

I had to import Symbol and Array.from polyfills into my app to get around this, but those are causing their own IE11 issues as well, just fewer than this caused.

**Comment by qmegas** [comment]
Because of that bug I had to stop using `for ... of` loop. The original report was made on Mar 19, 2018, 2 years have passed and still no fix.

**Comment by nicolo-ribaudo** [comment]
This report is not about `for ... of`; it's about array rest/spread. Anyway, I'm working on a fix for both which is smaller than the PR which was discarded: it only works with array and array-like objects. This should make it easier to land it.

**Comment by mikesherov** [comment]
@nicolo-ribaudo just wanted to say thank you so much for finally fixing this. Any idea when it may land in a stable version of Babel?   My use case is indeed the `rest spread` use case, and the assumeArray worked for me (mostly), but I'd like to be able to get rid of transforming for-ofs this way.

**Comment by nicolo-ribaudo** [comment]
Hi Mike!

You can follow the PRs in https://github.com/babel/babel/projects/12 for updates.
Some of them will already be included in the next release (on Friday), but the pending on the timing some of them could shift to the next patch (no more than 2 weeks) or minor (about 2 months).

If you have any comments about those PRs, please let us know!

**Comment by nicolo-ribaudo** [comment]
This is fixed for all the iterables defined in the ECMAScript spec by https://github.com/babel/babel/pull/11263 and https://github.com/babel/babel/pull/11285

**Comment by mikesherov** [comment]
> We are still working on adding opt-in support for array-like objects, so that it can be used with DOM lists.

Would this also work for jQuery? My other use case is:

```
const [elem] = $('div);
```

**Comment by cbratschi** [comment]
In which Babel release will this be shipped? So far I still see the failing version if assumeArray is false.

**Comment by nicolo-ribaudo** [comment]
Objects defined as iterable by the spec are supported starting from Babel 7.9.0. Array-like objects (which are not iterable, and technically shouldn't work with `for-of`) will be supported in 7.10.0.

**Comment by ljharb** [comment]
@nicolo-ribaudo fwiw arraylike objects don't need to work, just arrays, Maps, Sets, and strings.

**Comment by nicolo-ribaudo** [comment]
It should work with all the dom collections, that in old browsers are normal array-like objects. However, it will be opt-in since it's not spec-compliant.

**Comment by ljharb** [comment]
The DOM collections can't be identified beyond just by being arraylike? I'd expect their Object.prototype.toString to differ.
